### ISSUE TITLE
CrossOver Mac 21.1.0 Discussion

### ISSUE THREAD

**Issue by seathasky** [seathasky/FF14-MAC_ModSupport#41]
Discuss any performance related things with the latest CX update here.

**Comment by RussellAnas** [comment]
Macbook: M1 Max 16" 32 GB
DXVK: 1.9.2
MoltenVK: 1.1.4

I was on 21.0 before, and experienced a ton of lagginess (despite high fps, looked like game was less than 30fps instead) and stuttering (game freezes).

Jumped up to 21.1 and all lagginess and stuttering is gone. It actually looks like a game that performs around 80-100 fps. ProMotion is working now too; game looks super smooth. 

@seathasky do you think it's worth trying out patched 1.1.6 again?

**Comment by seathasky** [comment]
> Jumped up to 21.1 and all lagginess and stuttering is gone. It actually looks like a game that performs around 80-100 fps. ProMotion is working now too; game looks super smooth.
> 
> @seathasky do you think it's worth trying out patched 1.1.6 again?

That's good to hear! I haven't tested yet myself but plan on doing that this morning.

You are welcome to test 1.1.6 and see if that helps at all, just use Fleet's compile in public testing, or re-run our custom script and bypass the crosstie.

**Comment by seathasky** [comment]
Hackintosh: i7 10700k, AMD 5700 XT, 32gb DDR4 @ 3200mhz
DXVK: 1.9.2
MoltenVK: 1.1.6
In-Game Settings: MAX Settings (turned off tessellation, it drops fps)

Noticing minor increases on CX21.1

Getting around 180-200 fps in my usual testing spot with plugins loaded and Gshade enabled.

![Screen Shot 2021-11-30 at 8 00 58 AM](https://user-images.githubusercontent.com/17598024/144052225-c236ff6e-1356-489e-8253-1c2da21fd048.png)

Which is about a 5-10 fps difference than previous CX.

**Comment by kailred** [comment]
MacBook Air M1 2020
dvxk 1.9.2
moltenvk 1.1.6 patched.

updated to 21.1, changed no other settings and very minute stuttering has been introduced into normal gameplay. no performance increase.  will try others versions of dvxk and moltenvk to find best results for the mbair.

40-50fps across the board.

**Comment by cplr** [comment]
I'm having an interesting problem that I believe happened only after updating to CrossOver 21.1 (hence my posting of it here). In the XIVLauncherWine program, when I try to enter text into the password field, it crashes and throws an error window (the error window only has a title bar and no actual window underneath). I can log in using the default launcher, but unfortunately the XIVLauncher is not usable because of this. Anyone else have this? I don't see it mentioned anywhere...

**Comment by seathasky** [comment]
If you are using default launcher it will. Check out troubleshooting section in the guide. All the way at bottom

**Comment by seathasky** [comment]
> I'm having an interesting problem that I believe happened only after updating to CrossOver 21.1 (hence my posting of it here). In the XIVLauncherWine program, when I try to enter text into the password field, it crashes and throws an error window (the error window only has a title bar and no actual window underneath). I can log in using the default launcher, but unfortunately the XIVLauncher is not usable because of this. Anyone else have this? I don't see it mentioned anywhere...

Haven't seen that one yet, you could try reinstalling XIVlauncher. Or redo your bottle, just drag your SE folder over to the new bottle afterwards.

**Comment by zoeyronain** [comment]
> MacBook Air M1 2020 dvxk 1.9.2 moltenvk 1.1.6 patched.
> 
> updated to 21.1, changed no other settings and very minute stuttering has been introduced into normal gameplay. no performance increase. will try others versions of dvxk and moltenvk to find best results for the mbair.
> 
> 40-50fps across the board.

Same issue here with 21.1 on my M1 Pro 14" mbp. 

Also noticed I can't run it on maximum at 60-70fps anymore like with the macOS version, I'm hovering around 50 on high.  I did everything as stated including BUYING crossover thinking it may have been a bug with the trial.

**Comment by marzent** [comment]
If the native client is running faster than 21.1 then there is definitely sth wrong…
The only instance of that happening is with an old nvidia gpu and can be fixed by downgrading to MVK 1.1.0. 

Just to clarify the change happened going from 21.0 to 21.1 and .0 was faster than native and .1 is slower?

**Comment by zoeyronain** [comment]
> If the native client is running faster than 21.1 then there is definitely sth wrong… The only instance of that happening is with an old nvidia gpu and can be fixed by downgrading to MVK 1.1.0.
> 
> Just to clarify the change happened going from 21.0 to 21.1 and .0 was faster than native and .1 is slower?

I installed 21.1 out of the box, I just got my macbook. 

I followed [#25 (comment)](https://github.com/seathasky/FF14-MAC_ModSupport/issues/25#issue-1059037496) and at least at the menu it's 25-30fps higher than it was (73fps vs 48-53ish). (currently stuck there with a 2002 error)

Also just upgraded moltenvk and DXVK, we'll see how it goes.

**Comment by FleetAdmiralButter** [comment]
Hey @ZoeyCluff,

Depending on which build of MoltenVK you installed (1.1.4, Gcenx's 1.1.6, or my 1.1.6), some environment variables can be set to improve performance on M1 GPUs.

Let us know how it goes for you - you can either reply on this thread or ask for help on our Discord.

**Comment by zoeyronain** [comment]
Yeah, I just installed your 1.1.6 MoltenVK build. I keep getting dropped and winding up back in the the high 3000s queue over and over.

**Comment by nrylov** [comment]
Mac mini M1 8 GB
DXVK: 1.9.2
MoltenVK: 1.1.6

I just came across this yesterday and decided to try it out when I'm not using my windows laptop. I got the trial version of crossover 21 and followed all steps of the guide and the performance isn't quite what I was hoping it would be. When I make the game windowed with 1366x768 resolution, I get a pretty consistent ~60-65 fps. However, my monitor is 2560x1440, and that gets a consistent 25-30 fps. I also tried some other resolutions in between such as 1920x1080 and the fps was also around 25-30, even with the game's settings turned down to a minimum. I also tried playing around with the dxvk (patched, unpatched, async on/off) and moltenvk versions (option 2 from the installer, 1.1.6_2, 1.1.5) 

Is this expected performance for the M1? I haven't tried the regular mac version yet since I have the windows version but I'm considering making a new trial account to see how it performs.

**Comment by FleetAdmiralButter** [comment]
Hi @nrylov1,

In my testing, you should use DXVK 1.9.2 w/ async + Method 1 from here - https://github.com/seathasky/FF14-MAC_ModSupport/issues/16#issuecomment-984445292 for best performance. If you have trouble, you can try using version 1.1.4 of MoltenVK as it is more stable.

However, it is important to note that the M1 is quite a step down in terms of performance from the M1 Pro/Max, which most people here seem to have. A full 16-core M1 Pro performs about well as a GTX 1650 in gaming workloads, and you can expect about half of that from a regular M1. FF14 is also quite CPU heavy so the loss of 4 Performance cores also makes a difference.

**Comment by nrylov** [comment]
@FleetAdmiralButter 
That makes sense. It does seem that the CPU utilization is not the bottleneck as the utilization is not that high though the GPU utilization is much higher, but I'm not too familiar with how x86 emulation works on the M1. 

I'll give the method a try. If it doesn't end up significantly increasing fps it'll be a bit disappointing since wow runs perfectly at 100 fps for me, but it's understandable

Thanks

**Comment by FleetAdmiralButter** [comment]
@nrylov1, I don't think World of Warcraft is the best comparison here - for one, its a native app so you're not losing performance from DirectX -> Metal translation. The WoW engine is also a lot older (you can practically count the polygons on character models) so its easier to run too.

Good luck though! I hope this improves things for you.

**Comment by nrylov** [comment]
@FleetAdmiralButter unfortunately not much changed using the different settings for CX. I tested the official mac client and the performance is much better, can play fullscreen with ~50 fps in cities and I assume that would be higher in calmer areas. What would Rosetta be doing differently than CX?

**Comment by FleetAdmiralButter** [comment]
@nrylov1, both the official Mac client and Crossover were developed by Codeweavers and are very similar. However, the SE is using a very old version of Crossover and ships outdated versions of Wine, DXVK and MVK.

Since FFXIV is an x86 Windows app, both the Mac client and Crossover run under Rosetta. Wine itself depends on Rosetta to help translate x86 Windows instructions to ARM.

It is very weird that the Crossover bottle from here is performing worse. The only thing I can think of is your graphics settings. Have you checked that the "Tessellation" setting is set to Normal? Setting this to High can have really bad performance impacts.

**Comment by seathasky** [comment]
@nrylov1 Just like @FleetAdmiralButter said, the official client is behind by miles!

![Screen_Shot_2021-12-03_at_12 14 52_PM](https://user-images.githubusercontent.com/17598024/144746509-99ab63de-c73f-4d7b-af69-0d26a893288c.png)

This is the official client through the DXVK_HUD environment variable. You can see the official client is on 1.0.1 version of DXVK, which was first released around March 2019. It is beyond outdated.

**Comment by mixxorz** [comment]
Hiya.

I'm on an iMac 2020 with a Radeon Pro 5300.
DXVK 1.9.2
MVK 1.1.6

I had an issue where the FPS counter would display a solid 60+ FPS, but the game felt very jittery. After some tweaking, I found out that if I set my Frame Rate to 30FPS, the game looks _much_ smoother. The FPS counter still shows 60+ FPS.

On "None" or "60 fps" The DXVK FPS counter reads 60+ but the screen looks very jittery.

**Comment by seathasky** [comment]
> Hiya.
> 
> I'm on an iMac 2020 with a Radeon Pro 5300.
> 
> I had an issue where the FPS counter would display a solid 60+ FPS, but the game felt very jittery. After some tweaking, I found out that if I set my Frame Rate to 30FPS, the game looks _much_ smoother. The FPS counter still shows 60+ FPS.
> 
> On "None" or "60 fps" The DXVK FPS counter reads 60+ but the screen looks very jittery.

What version of MoltenVK are you using?

**Comment by mixxorz** [comment]
I've updated my comment but I'm running:

DXVK 1.9.2
MVK 1.1.6

**Comment by seathasky** [comment]
Try 1.1.4 or lower versions.

**Comment by quangxdu** [comment]
Has anyone noticed that subsequent launches of ffxiv after sleeping cause significant drops if FPS.  I'm seeing my average 100 fps drop to 30 on the same settings until a reboot. m1 max 14 inch 32 GB ram.

**Comment by mixxorz** [comment]
> Try 1.1.4 or lower versions.

Now trying MVK 1.1.4 and am seeing the same issue.

If it makes any difference I'm on Borderless Windowed mode.

It's not a big problem for me because as I said if I set it to 30 FPS it's super smooth. Like 60fps smooth. Just wanted to report that this is happening.

**Comment by realAbitbol** [comment]
> > Try 1.1.4 or lower versions.
> 
> Now trying MVK 1.1.4 and am seeing the same issue.
> 
> If it makes any difference I'm on Borderless Windowed mode.
> 
> It's not a big problem for me because as I said if I set it to 30 FPS it's super smooth. Like 60fps smooth. Just wanted to report that this is happening.

I have the very same issue with Radeon pro 580 in a 2017 iMac.
Try the 1.1.0 MoltenVK, it’s perfect for me and I think it will be the same for you

**Comment by lilyball** [comment]
@FleetAdmiralButter

> Depending on which build of MoltenVK you installed (1.1.4, Gcenx's 1.1.6, or my 1.1.6), some environment variables can be set to improve performance on M1 GPUs.

I'm using your 1.1.6 on a 16" M1 Max. Are there environment variables I should be setting (and how)?

**Comment by lilyball** [comment]
Alternatively, should I be installing Gcenx's 1.1.6? I see the install script was [updated a few days ago](https://github.com/marzent/ffxiv-on-mac/commit/ae40622c3f4ca1b60c5d38f4d3a1800733223c97) to suggest just Gcenx's, but I don't know if there's a benefit to switching now, or even what the right way to do so is (the instructions just talk about installing, not about updating).

**Comment by seathasky** [comment]
@lilyball Hey there lily, you want to use this release instead as it is the latest version of 1.1.6 and now included in our script sorry for the confusion. https://github.com/Gcenx/MoltenVK/releases

**Comment by lilyball** [comment]
@seathasky Is there a canonical way to apply updates like this, or do I need to go look at the install script and overwrite the files manually?

**Comment by seathasky** [comment]
Just use our script, and skip the crosstie to update

**Comment by FleetAdmiralButter** [comment]
Hey @lilyball - if you run our install script and select option 1) it should select and automatically pull in the latest release of MoltenVK from https://github.com/Gcenx/MoltenVK/releases. The latest patch sets up the most appropriate hardware sync method for your GPU (Apple/AMD/Nvidia).

**Comment by lilyball** [comment]
I just copied the file over by hand 😅 looking at the script, it does appear to unconditionally overwrite the `libMoltenVK.dylib.bak` file, meaning if I had used it to update, I would no longer have the original `libMoltenVK.dylib` file that came with CrossOver in the event that I decided to revert it.

**Comment by seathasky** [comment]
We include the original in the script from current stable Crossover so yes, you wouldn't have yours but you technically wouldn't lose it

**Comment by lilyball** [comment]
The script offers to reinstall the version from 21.0.0, but the current version is 21.1.0. I have no idea if libMoltenVK.dylib changed at all in 21.1.0 though.

**Comment by FleetAdmiralButter** [comment]
@lilyball, I forgot to mention earlier - if you use option 1) to pull in the latest libMoltenVK from Gcenx, you no longer need to set any MoltenVK-related environment variables because -

> The latest patch sets up the most appropriate hardware sync method for your GPU (Apple/AMD/Nvidia).

Also mentioning it again in case anyone stumbles across this again in the future :)

I don't believe there are any updates to either DXVK or MoltenVK in the update from 21.0 to 21.1 - this update mostly focused on Wine updates and compatibility fixes.

**Comment by seathasky** [comment]
The moltenvk version from 21.0 and 21.1 are exactly the same @lilyball

**Comment by marzent** [comment]
> I just copied the file over by hand 😅 looking at the script, it does appear to unconditionally overwrite the `libMoltenVK.dylib.bak` file, meaning if I had used it to update, I would no longer have the original `libMoltenVK.dylib` file that came with CrossOver in the event that I decided to revert it.

This is not true though the -n flag prevents overriding, your original backup is always safe

**Comment by marzent** [comment]
In fact if the script failed at any point all backups will always be save the way it’s setup

**Comment by justcolin** [comment]
**Computer:** M1 Max 32 GB (14" MBP)
** DXVK:** 1.9.2
**MVK:** 1.1.6 from the latest installer

**Comment by front-endian** [comment]
**Computer:** M1 Max 32 GB (14" MBP)
**DXVK:** 1.9.2 with async patches enabled
**MVK:** 1.1.6 from the latest installer

Framerates have been incredible (well above 60fps average, often stays around 100–120), but I'm getting random stutters where the whole game freezes, and audio stops for a split second every so often. Haven't really noticed a pattern to it.

Tried a few library versions, but haven't had luck fixing it yet and not sure how to figure out what the culprit is. (Not sure what I _should_ be swapping to address the issue, so just trying random things). Is there something to look for in the DXVK HUD to see why the freezes are happening? (Happy to split this out into a separate issue too.)

**Comment by front-endian** [comment]
Downgrading to MVK 1.1.4 dropped the frame rate to around 80 in most areas, but the short freezes seem to be rarer now, though still getting a couple.

**Comment by seathasky** [comment]
> Downgrading to MVK 1.1.4 dropped the frame rate to around 80 in most areas, but the short freezes seem to be rarer now, though still getting a couple.

Short freezes are due to on the fly shader caching most likely. Eventually you should build your cache up enough to stop the small freezes. I would recommend giving 1.1.6 a chance until you've played a bit longer or grab the cache file Marzent posted in support in our discord, I will pin it for you to download.

**Comment by front-endian** [comment]
Gotcha, I'll reload 1.1.6 tomorrow with that cache and see if it helps. Thank you!
